### DISCUSSION TITLE
Is there a way to suppress warnings but only for a given directory in Sphinx

### DISCUSSION THREAD

**Discussion by menelik003** [sphinx-doc/sphinx#12834] [Q&A]
Hi, I am generating Api documentation using `autoapi.extension` in sphinx. It is a large repo, and not all of the doc strings are well formatted and this generates a lot of sphinx docutils warnings. However I also want sphinx to fail on warnings for the rst files that are manually written. The autogenerated api gets written to a specific directory, so I was wondering if there was a way to ignore warnings just in that specific section

**Comment by jdillard** [comment]
You'd likely have to allow warnings and then make a custom CI script that fails the build if anything outside that directory fails.

I don't recommend this method as it is hard to crawl out of when you are ready to get to zero warnings. It's best to try and keep it at zero from the beginning, even if it seems more difficult in the beginning.
